### PR TITLE
RMB-601 PgUtil futurisation for Vert.x 4

### PR DESCRIPTION
**Pull request by MikhailFokanov**

### CODE REVIEW

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java [hunk 1]
```diff
@@ -397,53 +398,71 @@ static Method respond422method(Class<? extends ResponseDelegate> clazz) {
...
       }
+
+      Promise<Response> promise = Promise.promise();
+
       PostgresClient postgresClient = PgUtil.postgresClient(vertxContext, okapiHeaders);
       postgresClient.delete(table, id, reply -> {
         if (reply.failed()) {
-          asyncResultHandler.handle(response(table, id, reply.cause(), clazz, respond400, respond500));
-          return;
+          response(table, id, reply.cause(), clazz, respond400, respond500).onComplete(promise);
```
Comment (julianladisch) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java:427]: return;

is missing here and at similar places below.

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java [hunk 2]
```diff
@@ -659,12 +678,12 @@ private static void streamTrailer(HttpServerResponse response, ResultInfo result
           response.setStatusCode(400);
           response.putHeader(HttpHeaders.CONTENT_TYPE, "text/plain");
           response.end(message);
-          return;
```
Comment (julianladisch) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java:662]: return;

should be kept.

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java (hunk 1 above)
Comment (MikhailFokanov) [reply domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java:427]: Thanks!

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java (hunk 2 above)
Comment (MikhailFokanov) [reply domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java]: fixed

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java [hunk 3]
```diff
@@ -932,37 +1019,59 @@ public static void delete(String table,
         + clazz.getCanonicalName());
   }
 
-  /**
+    /**
```
Comment (julianladisch) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java:1022]:
```suggestion
  /**
```

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java [hunk 4]
```diff
@@ -932,37 +1019,59 @@ public static void delete(String table,
...
    * Post entity to table.
    *
    * <p>Create a random UUID for id if entity doesn't contain one.
    *
    * <p>All exceptions are caught and reported via the asyncResultHandler.
-   *
-   * @param table  table name
+   *  @param table  table name
```
Comment (julianladisch) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java:1028]:
```suggestion
   * @param table  table name
```

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java [hunk 5]
```diff
@@ -932,37 +1019,59 @@ public static void delete(String table,
...
    * @param entity  the entity to post. If the id field is missing or null it is set to a random UUID.
    * @param okapiHeaders  http headers provided by okapi
    * @param vertxContext  the current context
    * @param clazz  the ResponseDelegate class generated as defined by the RAML file, must have these methods:
-   *               headersFor201(), respond201WithApplicationJson(T, HeadersFor201),
-   *               respond400WithTextPlain(Object), respond500WithTextPlain(Object).
+*               headersFor201(), respond201WithApplicationJson(T, HeadersFor201),
+*               respond400WithTextPlain(Object), respond500WithTextPlain(Object).
```
Comment (julianladisch) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java:1034]:
```suggestion
*                  headersFor201(), respond201WithApplicationJson(T, HeadersFor201),
*                  respond400WithTextPlain(Object), respond500WithTextPlain(Object).
```

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java [hunk 6]
```diff
@@ -932,37 +1019,59 @@ public static void delete(String table,
...
+    post(table,entity,okapiHeaders,vertxContext,clazz).onComplete(asyncResultHandler);
+  }
+
+  /**
+   * Post entity to table.
+   *
+   * <p>Create a random UUID for id if entity doesn't contain one.
+   *
+   * <p>All exceptions are caught and reported via the asyncResultHandler.
+   *  @param table  table name
```
Comment (julianladisch) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java:1053]:
```suggestion
   * @param table  table name
```

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java [hunk 7]
```diff
@@ -983,25 +1092,27 @@ public static void delete(String table,
...
     } catch (Exception e) {
       logger.error(e.getMessage(), e);
-      asyncResultHandler.handle(response(e.getMessage(), respond500, respond500));
+      return response(e.getMessage(), respond500, respond500);
     }
   }
 
-  /**
+
+   /**
```
Comment (julianladisch) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java:1115]:
```suggestion
  /**
```

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java [hunk 8]
```diff
@@ -1014,21 +1125,41 @@ public static void delete(String table,
    * @param clazz  the ResponseDelegate class created from the RAML file with these methods:
    *               respond204(), respond400WithTextPlain(Object), respond404WithTextPlain(Object),
    *               respond409WithTextPlain(Object), respond500WithTextPlain(Object).
-   * @param asyncResultHandler  where to return the result created by clazz
+   * @return asyncResultHandler  where to return the result created by clazz
```
Comment (julianladisch) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java:1128]:
```suggestion
   * @param asyncResultHandler  where to return the result created by clazz
```

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java [hunk 9]
```diff
@@ -1084,31 +1216,48 @@ private static Method getRespond409(Class<? extends ResponseDelegate> clazz) {
 
   /**
    * Post a list of T entities to the database. Fail all if any of them fails.
-
    * @param table database table to store into
    * @param entities  the records to store
    * @param maxEntities  fail with HTTP 413 if entities.size() > maxEntities to avoid out of memory;
-   *     suggested value is from 100 ... 10000.
+ *     suggested value is from 100 ... 10000.
```
Comment (julianladisch) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java:1222]:
```suggestion
 *       suggested value is from 100 ... 10000.
```

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java [hunk 10]
```diff
@@ -1084,31 +1216,48 @@ private static Method getRespond409(Class<? extends ResponseDelegate> clazz) {
...
    * @param upsert  true to update records with the same id, false to fail all entities if one has an existing id
    * @param okapiHeaders  http headers provided by okapi
    * @param vertxContext  the current context
    * @param responseClass  the ResponseDelegate class created from the RAML file with these methods:
-   *               respond201(), respond409WithTextPlain(Object), respond413WithTextPlain(Object), respond500WithTextPlain(Object).
-   * @param asyncResultHandler  where to return the result created by responseClass
+*               respond201(), respond409WithTextPlain(Object), respond413WithTextPlain(Object), respond500WithTextPlain(Object).
```
Comment (julianladisch) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java:1227]:
```suggestion
  *               respond201(), respond409WithTextPlain(Object), respond413WithTextPlain(Object), respond500WithTextPlain(Object).
```

**Comment by MikhailFokanov** [comment]
@julianladisch I merged all you code style suggestions.

**Review by julianladisch — APPROVED** [review]
Looks good, thanks!

**Pull request merged**